AppController/lib/monit_interface.rb

```diff
@@ -116,27 +124,39 @@ def self.write_monit_config(watch, start_cmd, stop_cmd, port,
       }
     end
 
-    logfile = "/var/log/appscale/#{watch}-#{port}.log"
+    if port.nil?
+      new_start_cmd = "#{start_cmd}"
+      new_stop_cmd = "#{stop_cmd}"
+      new_match_cmd = "#{match_cmd}"
+      suffix = ""
+    else
+      new_start_cmd = "#{start_cmd} -p #{port}"
+      new_stop_cmd = "#{stop_cmd} #{port}"
+      new_match_cmd = "#{new_start_cmd}" if match_cmd == start_cmd
```

Comment:
If `port` is not nil, and `match_cmd` != `start_cmd`, `new_match_cmd` will not be defined.